src/tbx/mod.rs

```diff
@@ -0,0 +1,410 @@
+    pub fn from_path<P: AsRef<Path>>(path: P) -> Result<Self, TabixReaderPathError> {
+        if let Some(p) = path.as_ref().to_str() {
+            Ok(try!(Self::new(p.as_bytes())))
+        } else {
+            Err(TabixReaderPathError::InvalidPath)
+        }
+    }
+
+    pub fn from_url(url: &Url) -> Result<Self, TabixReaderError> {
+        Self::new(url.as_str().as_bytes())
+    }
+
+    /// Create a new TabixReader.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - the path.
+    fn new(path: &[u8]) -> Result<Self, TabixReaderError> {
+        let path = ffi::CString::new(path).unwrap();
+        let hts_file =
+            unsafe { htslib::hts_open(path.as_ptr(), ffi::CString::new("r").unwrap().as_ptr()) };
+        let hts_format = unsafe { (*htslib::hts_get_format(hts_file)).format };
+        let tbx = unsafe { htslib::tbx_index_load(path.as_ptr()) };
+        let mut header = Vec::new();
+        let mut buf = htslib::kstring_t {
+            l: 0,
+            m: 0,
+            s: ptr::null_mut(),
+        };
+        unsafe {
+            while htslib::hts_getline(hts_file, KS_SEP_LINE, &mut buf) >= 0 {
+                if buf.l > 0 && (*buf.s) as i32 == (*tbx).conf.meta_char {
+                    header.push(String::from(ffi::CStr::from_ptr(buf.s).to_str().unwrap()));
+                } else {
+                    break;
+                }
+            }
+        }
+
+        let mut nseq: i32 = 0;
+        let seqs = unsafe { htslib::tbx_seqnames(tbx, &mut nseq) };
+        let mut seqnames = Vec::new();
+        for i in 0..nseq {
+            unsafe {
+                seqnames.push(String::from(
+                    ffi::CStr::from_ptr(*seqs.offset(i as isize))
+                        .to_str()
+                        .unwrap(),
+                ));
+            }
+        }
+        unsafe {
+            libc::free(seqs as (*mut libc::c_void));
+        };
+
+        if tbx.is_null() {
+            Err(TabixReaderError::InvalidIndex)
+        } else {
+            Ok(TabixReader {
+                header,
+                seqnames,
+                hts_file,
+                hts_format,
+                tbx,
+                buf,
+                itr: None,
+                tid: -1,
+                beg: -1,
+                end: -1,
+            })
+        }
+    }
+
+    /// Get sequence ID from sequence name.
+    pub fn seq_name_to_id(&self, name: &str) -> Result<u32, SequenceLookupError> {
+        // TODO: naming?
+        let res = unsafe {
+            htslib::tbx_name2id(
+                self.tbx,
+                ffi::CString::new(name.as_bytes()).unwrap().as_ptr(),
+            )
+        };
+        if res < 0 {
+            Err(SequenceLookupError::Some)
+        } else {
+            Ok(res as u32)
+        }
+    }
+
+    /// Fetch region given by numeric sequence number and 1-based begin and end position.
+    pub fn fetch(&mut self, tid: u32, beg: u32, end: u32) -> Result<(), FetchError> {
```

Comment:
The rust name for these should be `start` and `end`, similar to the attributes of `std::ops::Range`. The indexes should be 0-based in order to be coherent with the rest of htslib (e.g., pos in bam::Record).

Reply:
OK, they already were 0-based, adjusted docs.